src/linked_list.rs

```diff
@@ -84,62 +83,64 @@ impl<T: ?Sized> Wrapper<T> for &T {
 }
 
 /// A descriptor of wrapped list elements.
-pub trait AdapterWrapped: Adapter {
+pub trait GetLinksWrapped: GetLinks {
     /// Specifies which wrapper (e.g., `Box` and `Arc`) wraps the list entries.
     type Wrapped: Wrapper<Self::EntryType>;
 }
 
-impl<T: ?Sized> AdapterWrapped for Box<T>
+impl<T: ?Sized> GetLinksWrapped for Box<T>
 where
-    Box<T>: Adapter,
+    Box<T>: GetLinks,
 {
-    type Wrapped = Box<<Box<T> as Adapter>::EntryType>;
+    type Wrapped = Box<<Box<T> as GetLinks>::EntryType>;
 }
 
-unsafe impl<T: Adapter + ?Sized> Adapter for Box<T> {
+impl<T: GetLinks + ?Sized> GetLinks for Box<T> {
     type EntryType = T::EntryType;
-
-    #[inline]
-    fn to_links(data: &Self::EntryType) -> &Links<Self::EntryType> {
-        <T as Adapter>::to_links(data)
+    fn get_links(data: &Self::EntryType) -> &Links<Self::EntryType> {
+        <T as GetLinks>::get_links(data)
     }
 }
 
-impl<T: ?Sized> AdapterWrapped for Arc<T>
+impl<T: ?Sized> GetLinksWrapped for Arc<T>
 where
-    Arc<T>: Adapter,
+    Arc<T>: GetLinks,
 {
-    type Wrapped = Arc<<Arc<T> as Adapter>::EntryType>;
+    type Wrapped = Arc<<Arc<T> as GetLinks>::EntryType>;
 }
 
-unsafe impl<T: Adapter + ?Sized> Adapter for Arc<T> {
+impl<T: GetLinks + ?Sized> GetLinks for Arc<T> {
     type EntryType = T::EntryType;
 
     #[inline]
-    fn to_links(data: &Self::EntryType) -> &Links<Self::EntryType> {
-        <T as Adapter>::to_links(data)
+    fn get_links(data: &Self::EntryType) -> &Links<Self::EntryType> {
+        <T as GetLinks>::get_links(data)
     }
 }
 
 /// A linked list.
 ///
 /// Elements in the list are wrapped and ownership is transferred to the list while the element is
 /// in the list.
-pub struct List<G: AdapterWrapped> {
-    list: unsafe_list::List<G>,
+pub struct List<G: GetLinksWrapped> {
+    list: RawList<G>,
 }
 
-impl<G: AdapterWrapped> List<G> {
+impl<G: GetLinksWrapped> List<G> {
     /// Constructs a new empty linked list.
     pub const fn new() -> Self {
         Self {
-            list: unsafe_list::List::new(),
+            list: RawList::new(),
         }
     }
 
+    /// Returns an iterator for the list starting at the first entry.
+    pub fn iter(&self) -> Iterator<'_, G> {
+        Iterator::new(self)
+    }
+
     /// Returns whether the list is empty.
-    #[inline]
-    pub const fn is_empty(&self) -> bool {
+    pub fn is_empty(&self) -> bool {
```

Comment:
Why not use const fn?

Reply:
In prev version : `use crate::unsafe_list::{self, Adapter, Cursor, Links}; `   function param  `&self` not means the  ref of instances , otherwise it is unsafe_list::self type ; 

Current, `&self`  not a type, it is RawList instances ref